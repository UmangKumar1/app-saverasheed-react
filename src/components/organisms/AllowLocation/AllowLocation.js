import React, { Component } from 'react'
import { css } from 'react-emotion'
import PropTypes from 'prop-types'

import Modal from '../../molecules/Modal'
import Button from '../../atoms/Button'
import { ModalP, ModalTitle } from '../../molecules/Modal/Modal'

const { bool, func } = PropTypes

const buttonCx = css({
  fontSize: 18,
  padding: '8px 16px',
})

const screenEffectCx = css({
  display: 'flex',
  alignItems: 'center',
})

class AllowLocation extends Component {
  static propTypes = {
    allowLocation: func.isRequired,
    isLocationAllowed: bool,
  }

  render() {
    const { allowLocation, isLocationAllowed } = this.props
    return (
      <Modal on={!isLocationAllowed} screenEffectClassName={screenEffectCx}>
        <ModalTitle>Save Rasheed is a Geo dApp.</ModalTitle>
        <ModalP>You'll need to allow location access to play the game!</ModalP>
        <ModalP>Please allow location access below.</ModalP>
        <Button className={buttonCx} onClick={allowLocation}>
          Allow
        </Button>
      </Modal>
    )
  }
}

export default AllowLocation
