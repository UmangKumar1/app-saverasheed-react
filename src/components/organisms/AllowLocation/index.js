export { default, default as AllowLocation } from './AllowLocation'
