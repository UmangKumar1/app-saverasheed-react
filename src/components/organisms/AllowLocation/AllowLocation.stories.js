import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import AllowLocation from '.'

storiesOf('AllowLocation', module).add('default', () => (
  <AllowLocation allowLocation={action('allow!')} isLocationAllowed />
))
