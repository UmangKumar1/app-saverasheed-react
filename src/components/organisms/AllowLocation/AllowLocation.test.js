import React from 'react'
import { shallow } from 'enzyme'
import AllowLocation from '.'

const wrap = (props = {}) => shallow(<AllowLocation {...props} />)

describe('AllowLocation', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({
      allowLocation: jest.fn(),
      isLocationAllowed: true,
    })
    expect(wrapper).toMatchSnapshot()
  })
})
