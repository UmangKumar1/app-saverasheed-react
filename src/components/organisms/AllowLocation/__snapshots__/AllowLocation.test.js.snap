// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllowLocation renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllowLocation
    allowLocation={[MockFunction]}
    isLocationAllowed={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(h1)>
          Save Rasheed is a Geo dApp.
        </Styled(h1)>,
        <Styled(p)>
          You'll need to allow location access to play the game!
        </Styled(p)>,
        <Styled(p)>
          Please allow location access below.
        </Styled(p)>,
        <Styled(button)
          className="css-6yi3vl"
          onClick={[MockFunction]}
        >
          Allow
        </Styled(button)>,
      ],
      "on": false,
      "screenEffectClassName": "css-70qvj9",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Save Rasheed is a Geo dApp.",
        },
        "ref": null,
        "rendered": "Save Rasheed is a Geo dApp.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "You'll need to allow location access to play the game!",
        },
        "ref": null,
        "rendered": "You'll need to allow location access to play the game!",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Please allow location access below.",
        },
        "ref": null,
        "rendered": "Please allow location access below.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Allow",
          "className": "css-6yi3vl",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Allow",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(h1)>
            Save Rasheed is a Geo dApp.
          </Styled(h1)>,
          <Styled(p)>
            You'll need to allow location access to play the game!
          </Styled(p)>,
          <Styled(p)>
            Please allow location access below.
          </Styled(p)>,
          <Styled(button)
            className="css-6yi3vl"
            onClick={[MockFunction]}
          >
            Allow
          </Styled(button)>,
        ],
        "on": false,
        "screenEffectClassName": "css-70qvj9",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Save Rasheed is a Geo dApp.",
          },
          "ref": null,
          "rendered": "Save Rasheed is a Geo dApp.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "You'll need to allow location access to play the game!",
          },
          "ref": null,
          "rendered": "You'll need to allow location access to play the game!",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Please allow location access below.",
          },
          "ref": null,
          "rendered": "Please allow location access below.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Allow",
            "className": "css-6yi3vl",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Allow",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
